src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -364,6 +372,147 @@ private static bool IsSafeForReordering(BoundExpression expression, RefKind kind
             }
         }
 
+        private ImmutableArray<BoundExpression> VisitArguments(
+            SyntaxNode syntax,
+            ImmutableArray<BoundExpression> arguments,
+            Symbol methodOrIndexer,
+            ImmutableArray<int> argsToParamsOpt,
+            ImmutableArray<RefKind> argumentRefKindsOpt,
+            [NotNullIfNotNull("rewrittenReceiver")] ref BoundExpression? rewrittenReceiver,
+            out ArrayBuilder<LocalSymbol>? temps,
+            out bool receiverIsArgumentSideEffectSequence)
+        {
+            Debug.Assert(argumentRefKindsOpt.IsDefault || argumentRefKindsOpt.Length == arguments.Length);
+            var requiresInstanceReceiver = methodOrIndexer.RequiresInstanceReceiver() && methodOrIndexer is not MethodSymbol { MethodKind: MethodKind.Constructor };
+            Debug.Assert(!requiresInstanceReceiver || rewrittenReceiver != null || _inExpressionLambda);
+            temps = null;
+            var positionsAssignedToTemp = BitVector.Null;
+            receiverIsArgumentSideEffectSequence = false;
+
+            if (arguments.IsEmpty)
+            {
+                return arguments;
+            }
+
+            var visitedArgumentsBuilder = ArrayBuilder<BoundExpression>.GetInstance(arguments.Length);
+            var parameters = methodOrIndexer.GetParameters();
+
+            for (int i = 0; i < arguments.Length; i++)
+            {
+                var argument = arguments[i];
+                if (argument is BoundDiscardExpression discard)
+                {
+                    temps ??= ArrayBuilder<LocalSymbol>.GetInstance();
+                    visitedArgumentsBuilder.Add(_factory.MakeTempForDiscard(discard, temps));
+                    continue;
```

Comment:
>continue; [](http://example.com/codeflow?start=20&length=9)

It feels like we should be marking this argument as capured in a temp otherwise we are probably capturing it again if it is used by a builder. Also, we should make sure we have struct tests that we discussed offline around discard arguments

Reply:
> struct tests that we discussed offline around discard arguments

I'm not sure how you'd observe the results with a discard argument unless I was to use `Unsafe.SkipInit`. Whatever result we set in the builder constructor isn't observable in the body of the method, because the method itself must initialize the struct.

Reply:
VB, for example, doesn't respect out, it treats them as ref and allows reading them

Reply:
VB allows placing the <Out> attribute on the ByRef parameter what C# will see as out

Reply:
Ah, I see. I'll add a test like this.